sdk/storage/azure-storage-file-share/azure/storage/fileshare/_models.py

```diff
@@ -461,35 +636,52 @@ class Handle(DictMixin):
+    :keyword str session_id: SMB session ID in context of which the file handle was opened.
+    :keyword str client_ip: Client IP that opened the handle.
+    :keyword ~datetime.datetime open_time: Time when the session that previously opened
+        the handle has last been reconnected. (UTC)
+    :keyword ~datetime.datetime last_reconnect_time: Time handle was last connected to. (UTC)
     :keyword access_rights: Access rights of the handle.
-    :paramtype access_rights: Optional[List[Literal['Read', 'Write', 'Delete']]]
+    :paramtype access_rights: List[Literal['Read', 'Write', 'Delete']]
     """
 
-    def __init__(self, **kwargs):
-        self.client_name = kwargs.get('client_name')
-        self.id = kwargs.get('handle_id')
-        self.path = kwargs.get('path')
-        self.file_id = kwargs.get('file_id')
-        self.parent_id = kwargs.get('parent_id')
-        self.session_id = kwargs.get('session_id')
-        self.client_ip = kwargs.get('client_ip')
-        self.open_time = kwargs.get('open_time')
-        self.last_reconnect_time = kwargs.get('last_reconnect_time')
-        self.access_rights = kwargs.get('access_right_list')
+    client_name: str
+    """Name of the client machine where the share is being mounted."""
+    client_id: str
+    """XSMB service handle ID."""
+    path: str
+    """File or directory name including full path starting from share root."""
+    file_id: str
+    """FileId uniquely identifies the file or directory."""
+    parent_id: str
+    """ParentId uniquely identifies the parent directory of the object."""
+    session_id: str
+    """SMB session ID in context of which the file handle was opened."""
+    client_ip: str
+    """Client IP that opened the handle."""
+    open_time: "datetime"
+    """Time when the session that previously opened the handle was last been reconnected. (UTC)"""
+    last_reconnect_time: "datetime"
+    """Time handle that was last connected to. (UTC)"""
+    access_rights: List[Literal['Read', 'Write', 'Delete']]
+    """Access rights of the handle."""
+
+    def __init__(self, **kwargs: Any) -> None:
+        self.client_name = kwargs.get('client_name')  # type: ignore [assignment]
+        self.id = kwargs.get('handle_id')  # type: ignore [assignment]
+        self.path = kwargs.get('path')  # type: ignore [assignment]
+        self.file_id = kwargs.get('file_id')  # type: ignore [assignment]
+        self.parent_id = kwargs.get('parent_id')  # type: ignore [assignment]
+        self.session_id = kwargs.get('session_id')  # type: ignore [assignment]
+        self.client_ip = kwargs.get('client_ip')  # type: ignore [assignment]
+        self.open_time = kwargs.get('open_time')  # type: ignore [assignment]
+        self.last_reconnect_time = kwargs.get('last_reconnect_time')  # type: ignore [assignment]
+        self.access_rights = kwargs.get('access_right_list')  # type: ignore [assignment]
```

Comment:
Just to confirm, none of these are `Optional`? We get all these fields all the time? (That may be a difficult question to answer because `Handle` is a tough thing to test :) )

Reply:
Looks like `last_reconnect_time` was the only possible `Optional` field. In addition, there was a typo in `handle_id` which was supposed to be just `id`. Both are fixed.